dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -362,11 +362,13 @@ object DistriOptimizer {
     val comupteThresholdbatchSize = state.get[Int]("comupteThresholdbatchSize").get
     val models = dataset.originRDD().mapPartitions(_ => {
       val (broadcastModel, broadcastCriterion, broadcastState) = broadcast.value
-      if (checkSingleton) {
-        require(Engine.checkSingleton(), "Detect multi-task run on one Executor/Container. " +
-          "Currently not support this")
-      } else {
-        logger.info("Skip singleton check")
+      if (!Engine.checkSingleton()) {
+        if (checkSingleton) {
+          require(Engine.checkSingleton(), "Detect multi-task run on one Executor/Container. " +
```

Comment:
If there are more than one task run in an executor and checkSingleton is enable, we need throw exception here, the second checkSingleton is guaranteed to fail to trigger the exception